src/App.jsx

```diff
@@ -14,45 +17,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const ALPHABET_SORT = 'alphabet';
+const LENGTH_SORT = 'length';
+
+function getSortedGoods(goods, sortType) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case ALPHABET_SORT:
+          return good1.localeCompare(good2);
+
+        case LENGTH_SORT:
+          return good1.length - good2.length;
+
+        default:
+          return preparedGoods;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const visibleGoods = getSortedGoods(goodsFromServer, sortType);
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function handleReverse(goods) {
+    return [...goods].reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortType(ALPHABET_SORT)}
+          type="button"
+          className={cn(
+            'button', 'is-info', { 'is-light': sortType !== ALPHABET_SORT },
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortType(LENGTH_SORT)}
+          type="button"
+          className={cn(
+            'button', 'is-success', { 'is-light': sortType !== LENGTH_SORT },
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={cn('button', 'is-warning', { 'is-light': !isReversed })}
+        >
+          Reverse
+        </button>
+
+        <button
+          onClick={() => {
+            setSortType(''); setIsReversed(false);
+          }}
+          type="button"
+          className="button is-danger is-light"
+          style={{ display: isReversed || sortType ? 'block' : 'none' }}
```

Comment:
it's bad practice to add inline styles w/o necessary
use [conditional rendering](https://react.dev/learn/conditional-rendering) instead